Scripts/script-ExtractDomainFromUrlFormat.yml

```diff
@@ -7,25 +7,34 @@ script: |-
   from validate_email import validate_email
   import re
 
+  def extract_domain(the_input):
+      is_email = validate_email(the_input)
+      if is_email:
+          domain = re.search('@[\w.]+', the_input)
+          if len(domain.group()) > 0:
+              return(domain.group()[1:])
+      else:
+          domain = get_tld(the_input, fail_silently=True)
+          if not domain:
+              the_input_with_http = 'https://' + the_input
+              domain = get_tld(the_input_with_http, fail_silently=True)
+              if domain:
+                  return(domain)
+              else:
+                 return('')
+          else:
+              return(domain)
 
-  inp = demisto.args()['input']
+  domains = []
+  the_input = demisto.args().get('input')
 
-  is_email = validate_email(inp)
-  if is_email:
-      domain = re.search("@[\w.]+", inp)
-      if len(domain.group()) > 0:
-          demisto.results(domain.group()[1:])
-  else:
-      domain = get_tld(inp, fail_silently=True)
-      if not domain:
-          inp_with_http = 'https://'+inp
-          domain = get_tld(inp_with_http, fail_silently=True)
-          if domain:
-              demisto.results(domain)
-          else:
-             demisto.results("")
-      else:
-          demisto.results(domain)
+  if isinstance(the_input, str) or isinstance(the_input, unicode):
+      the_input = argToList(the_input)
+
+  # Otherwise assumes it's an array
+  for item in the_input:
+      domains.append(extract_domain(item))
+  demisto.results(domains)
```

Comment:
using results with the array is not good as it creates an entry for each element in the array. we want to return one array with all results. You should change to return a single entry of the form:
```
demisto.results({
      'Type': entryTypes['note'],
      'ContentsFormat': formats['json'],
      'Contents': domains,      
  })
```

Also note: when a single param is passed in (not array) we should return a single string in the Content value and not an array with a single string.